symfony/src/Controller/AccessManagementController.php

```diff
@@ -75,21 +77,25 @@ public function member($groupId, Request $request)
         if ($groupMemberForm->isSubmitted()) {
             if ($groupMemberForm->isValid()) {
                 $email = $groupMemberForm->get('email')->getData() . self::MEMBER_DOMAIN;
-                $result = $this->googleGroupsService->addMember($group->email, $email);
-                if ($result['status'] === 'success') {
-                    $this->addFlash('success', $result['message']);
-                    $this->loggerService->log(Log::GROUP_MEMBER_ADD, [
-                        'member' => $email,
-                        'result' => 'success'
-                    ]);
-                    return $this->redirectToRoute('access_manage_user_group', ['groupId' => $groupId]);
+                if ($this->googleGroupsService->getUser($email)) {
+                    $result = $this->googleGroupsService->addMember($group->email, $email);
+                    if ($result['status'] === 'success') {
+                        $this->addFlash('success', $result['message']);
+                        $this->loggerService->log(Log::GROUP_MEMBER_ADD, [
+                            'member' => $email,
+                            'result' => 'success'
+                        ]);
+                        return $this->redirectToRoute('access_manage_user_group', ['groupId' => $groupId]);
+                    }
+                    $errorMessage = isset($result['code']) && $result['code'] === 409 ? 'Member already exists.' : 'Error occurred. Please try again.';
+                } else {
+                    $errorMessage = 'User not registred.';
```

Comment:
Can we move this outside the isValid block, and then handle these as form errors tied to the username field?